compiler/rustc_hir_typeck/src/demand.rs

```diff
@@ -315,7 +315,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     probe::ProbeScope::TraitsInScope,
                     None,
                 ) {
-                    Ok(pick) => pick.self_ty,
+                    Ok(pick) => eraser.fold_ty(pick.self_ty),
```

Comment:
`eraser` is introduced as `// Hack to make equality checks on types with inference variables and regions useful.`.
Could you also add that it's used to make it not completely broken (or some other wording)?

Reply:
well, if you plan on deleting or more fundamentally fixing the code later then it doesn't matter that much